File: Lecture 1/mario.c
#include <stdio.h>
#include <cs50.h>

int main (void){
    int height = get_int("Enter Height\n");
    while (height<1||height>8){
        printf("Height must be between one and eight");
         height = get_int("Enter Height");
    }
    bool half;
    for (int i=0;i<height;i++){
        half = false;
        for(int j=0;j<height*2;j++){
            if (half==false) {
              if (j>=height-(i+1)){
                  printf("#");
              } else printf(" ");
            } else{
            if (j<=height+(i)) {
                printf("#");
            } else printf(" ");
            }
            if (j==height-1){
                half=true;
                printf("  ");
            }
        }
        printf("\n\n");
    }
}

File: Lecture 6/sentimental-mario-more/mario.py
def main():
    print("How many blocks?")
    height = get_num()
    while height<1 or height>8:
        print("Insert a number between one and eight inclusive")
        height = get_num()
    for i in range(height):
        half = False
        for j in range(height*2):
            if half==False:
                if j>=height-(i+1):
                    print("#", end="")
                else:
                   print(" ", end="")
            elif j<=height+(i):
                print("#", end="")
            if (j==height-1):
                half=True
                print("  ", end="")
        print()


def get_num():
    num = None
    while num is None:
        input_value = input()
        try:
            # try and convert the string input to a number
            num = int(input_value)
        except ValueError:
            # tell the user off
            print(f"{input_value} is not a number, please enter a number only")
    return num
if __name__ == '__main__':
    main()